oap-server/server-storage-plugin/storage-influxdb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/influxdb/InfluxClient.java

```diff
@@ -0,0 +1,194 @@
+     */
+    public static final String TAG_ENTITY_ID = "entity_id";
+
+    private final String database;
+
+    public InfluxClient(InfluxStorageConfig config) {
+        this.config = config;
+        this.database = config.getDatabase();
+    }
+
+    public final String getDatabase() {
+        return database;
+    }
+
+    @Override
+    public void connect() {
+        influx = InfluxDBFactory.connect(config.getUrl(), config.getUser(), config.getPassword(),
+            new OkHttpClient.Builder(), InfluxDB.ResponseFormat.MSGPACK);
+        influx.query(new Query("CREATE DATABASE " + database));
+
+        influx.setDatabase(database);
+        influx.enableBatch();
+    }
+
+    /**
+     * To get a connection of InfluxDB
+     *
+     * @return InfluxDB's connection
+     */
+    public InfluxDB getInflux() {
+        return influx;
+    }
+
+    /**
+     * Request with a {@link Query} to InfluxDB and return a set of {@link QueryResult.Result}s.
+     *
+     * @param query
+     * @return a set of Result.
+     * @throws IOException
+     */
+    public List<QueryResult.Result> query(Query query) throws IOException {
+        if (logger.isDebugEnabled()) {
+            logger.debug("SQL Statement: {}", query.getCommand());
+        }
+
+        try {
+            QueryResult result = getInflux().query(query);
+            if (result.hasError()) {
+                throw new IOException(result.getError());
+            }
+            return result.getResults();
+        } catch (Exception e) {
+            throw new IOException(e.getMessage() + System.lineSeparator() + "SQL Statement: " + query.getCommand(), e);
+        }
+    }
+
+    /**
+     * Request with one statement to InfluxDB and return a set of {@link QueryResult.Series}s.
+     *
+     * @param query
+     * @return a set of Series
+     * @throws IOException
+     */
+    public List<QueryResult.Series> queryForSeries(Query query) throws IOException {
+        return query(query).get(0).getSeries();
+    }
+
+    /**
+     * Data management, to drop a time-series by measurement and time-series name specified. If an exception isn't
+     * thrown, it means execution success.
+     *
+     * @param measurement
+     * @param timeBucket
+     * @throws IOException
+     */
+    public void dropSeries(String measurement, long timeBucket) throws IOException {
+        Query query = new Query("DROP SERIES FROM " + measurement + " WHERE time_bucket='" + timeBucket + "'");
+        QueryResult result = getInflux().query(query);
+
+        if (result.hasError()) {
+            throw new IOException("Statement: " + query.getCommand() + ", ErrorMsg: " + result.getError());
+        }
+    }
+
+    /**
+     * Data management, to delete data by a statement. If an exception isn't thrown, it means execution success.
+     *
+     * @param statement
+     * @throws IOException
+     */
+    public void queryForDelete(String statement) throws IOException {
```

Comment:
Don't see anyone uses this. Please confirm. And `queryForDelete` is not a good idea. For time-series data, it is better to delete by time bucket only.

Reply:
Yes, I will remove